src/Horse.Router.pas

```diff
@@ -142,8 +142,15 @@ procedure THorseRouterTree.ExecuteInternal(APath: TQueue<string>; AHTTPType: TMe
           begin
             if AResponse.Status = THTTPStatus.NotFound.ToInteger then
               AResponse.Send('');
-
-            LCallback.Items[LIndexCallback](ARequest, AResponse, LNext);
+            try
+              LCallback.Items[LIndexCallback](ARequest, AResponse, LNext);
+            except
+              on E: Exception do
```

Comment:
Isn't it better to generate an exception from Horse?
Tested this implementation using the Handle-Exception too? To see how it behaves?

Reply:
If you use the EHorseException check and the user does not handle the error and fires as an EHorseException, the error code propagation is't keep correct!

This is't to replace the HandleException, it's just a correction of the error propagation!